x-pack/packages/kbn-elastic-assistant/impl/assistant/conversations/conversation_settings_management/index.tsx

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiPanel, EuiSpacer, EuiConfirmModal, EuiInMemoryTable } from '@elastic/eui';
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { ActionTypeRegistryContract } from '@kbn/triggers-actions-ui-plugin/public';
+import { Conversation } from '../../../assistant_context/types';
+import { ConversationTableItem, useConversationsTable } from './use_conversations_table';
+import { ConversationStreamingSwitch } from '../conversation_settings/conversation_streaming_switch';
+import { AIConnector } from '../../../connectorland/connector_selector';
+import * as i18n from './translations';
+
+import { Prompt } from '../../types';
+import { ConversationsBulkActions } from '../../api';
+import { useAssistantContext } from '../../../assistant_context';
+import { useConversationDeleted } from '../conversation_settings/use_conversation_deleted';
+import { useFlyoutModalVisibility } from '../../common/components/assistant_settings_management/flyout/use_flyout_modal_visibility';
+import { Flyout } from '../../common/components/assistant_settings_management/flyout';
+import { CANCEL, DELETE } from '../../settings/translations';
+import { ConversationSettingsEditor } from '../conversation_settings/conversation_settings_editor';
+import { useConversationChanged } from '../conversation_settings/use_conversation_changed';
+import { DEFAULT_PAGE_INDEX, DEFAULT_PAGE_SIZE } from '../../settings/const';
+
+interface Props {
+  actionTypeRegistry: ActionTypeRegistryContract;
+  allSystemPrompts: Prompt[];
+  assistantStreamingEnabled: boolean;
+  connectors: AIConnector[] | undefined;
+  conversationSettings: Record<string, Conversation>;
+  conversationsSettingsBulkActions: ConversationsBulkActions;
+  conversationsLoaded: boolean;
```

Comment:
I'm not sure I understand this new `conversationsLoaded` flag being plumbed through all these components. Can we not use the `useFetchCurrentUserConversations()` hook in the specific components where this is needed? The use query hook caches so it should be no issue to use this hook to determine if the conversations have been loaded.

I also believe you can use something like [`queryChache.find()`](https://tanstack.com/query/v4/docs/reference/QueryCache#querycachefind) and use the `CONVERSATIONS_QUERY_KEYS` cache key to look up if anything has been retrieved.

Reply:
`conversationsLoaded` was added for https://github.com/elastic/kibana/pull/184678/files#diff-6d61cccd98ee658cead30501f82b0016e62e6dbc93531a64a17209c7ed04b32dR191-R196

The lines above was added because I found `conversationSettings` here sometimes contained only the `base conversations` without the `saved conversations` , therefore cause the problem whenever I had to reset the conversations.

The conversations between conversations tab and system prompts tab has to be aligned after every change. I guess that's the reason the conversations state is shared across tabs. Otherwise we might need to refetch conversation list every time switching on conversations tab or system prompt tab to get the latest state. In the current implementation, it refetches only when changes happened.

I'm worried if we use keys to find if conversations has been loaded might not be accurate, as it might not saying it's the latest conversation list is loaded.

Reply:
Ahhh, I see. Thank you for explaining! 🙂 

> Otherwise we might need to refetch conversation list every time switching on conversations tab or system prompt tab to get the latest state.
>
> I'm worried if we use keys to find if conversations has been loaded might not be accurate, as it might not saying it's the latest conversation list is loaded.

So this should be fine (to re-query conversations on each tab). We could then rely on the useQuery cache, and after changes are applied we can call `queryClient.invalidateQueries()`for the conversations key. We have [helper hooks for this for other hooks like the KB](https://github.com/elastic/kibana/blob/1b872fbf9dc90d3c82a569a9faef9e360fc41171/x-pack/packages/kbn-elastic-assistant/impl/assistant/api/knowledge_base/use_knowledge_base_status.tsx#L65-L80), but not conversations right now -- it's an easy add though.

---

I would favor the extra query for conversations when switching tabs over plumbing `conversationsLoaded` through all these components from a complexity standpoint, so if that works I say do that. But if it's more complicated we can leave as-is and clean this up when we have prompt persistence. Does that sound okay @angorayc? cc @YulNaumenko for visibility since you're working prompt peristence.

Reply:
Yup, Thank you for explaining! @spong  I'll create another PR to clean this up,